### PR TITLE
Add support for ppc64le

### PR DESCRIPTION
**Pull request by pavolloffay**
This PR contains:

- [ ] bugfix
- [ ] feature
- [ ] refactor
- [ ] documentation
- [x] other

Are tests included?

- [ ] yes (_bugfixes and features will not be merged without tests_)
- [x] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [ ] no

List any relevant issue numbers:

### Description

Add support for ppc64le linux gnu target.

The PR is similar to https://github.com/rollup/rollup/pull/5346. Heads up I am not working with node ecosystem. I just need to build https://github.com/jaegertracing/jaeger-ui on ppc64le which gives me at the moment:

```
#9 548.1 $ NODE_ENV=production REACT_APP_VSN_STATE=$(../../scripts/get-tracking-version.js) vite build
#9 553.5 /workspace/node_modules/rollup/dist/native.js:38
#9 553.5 	throw new Error(
#9 553.5 	      ^
#9 553.5 
#9 553.5 Error: Your current platform "linux" and architecture "s390x" combination is not yet supported by the native Rollup build. Please use the WASM build "@rollup/wasm-node" instead.
#9 553.5 
#9 553.5 The following platform-architecture combinations are supported:
#9 553.5 android-arm
#9 553.5 android-arm64
#9 553.5 darwin-arm64
#9 553.5 darwin-x64
#9 553.5 linux-arm
#9 553.5 linux-arm64
#9 553.5 linux-arm64 (musl)
#9 553.5 linux-riscv64
#9 553.5 linux-x64
#9 553.5 linux-x64 (musl)
#9 553.5 win32-arm64
#9 553.5 win32-ia32
#9 553.5 win32-x64
#9 553.5 
#9 553.5 If this is important to you, please consider supporting Rollup to make a native build for your platform and architecture available.
#9 553.5     at Object.<anonymous> (/workspace/node_modules/rollup/dist/native.js:38:8)
#9 553.5     at Module._compile (node:internal/modules/cjs/loader:1356:14)
#9 553.5     at Module._extensions..js (node:internal/modules/cjs/loader:1414:10)
#9 553.5     at Module.load (node:internal/modules/cjs/loader:1197:32)
#9 553.5     at Module._load (node:internal/modules/cjs/loader:1013:12)
#9 553.5     at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:202:29)
#9 553.5     at ModuleJob.run (node:internal/modules/esm/module_job:195:25)
#9 553.5     at async ModuleLoader.import (node:internal/modules/esm/loader:336:24)
#9 553.5 
#9 553.5 Node.js v18.19.0
#9 553.6 error Command failed with exit code 1.
#9 553.6 info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
#9 ERROR: process "/bin/sh -c yarn install --frozen-lockfile && cd ./packages/jaeger-ui && yarn build" did not complete successfully: exit code: 1
------
 > [4/4] RUN  yarn install --frozen-lockfile && cd ./packages/jaeger-ui && yarn build:
553.5     at Module._extensions..js (node:internal/modules/cjs/loader:1414:10)
553.5     at Module.load (node:internal/modules/cjs/loader:1197:32)
553.5     at Module._load (node:internal/modules/cjs/loader:1013:12)
553.5     at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:202:29)
553.5     at ModuleJob.run (node:internal/modules/esm/module_job:195:25)
553.5     at async ModuleLoader.import (node:internal/modules/esm/loader:336:24)
553.5 
553.5 Node.js v18.19.0
553.6 error Command failed with exit code 1.
553.6 info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
------
WARNING: No output specified with docker-container driver. Build result will only remain in the build cache. To push result image into registry use --push or to load image into docker use --load
Dockerfile:7
--------------------
   5 |     COPY . .
   6 |     
   7 | >>> RUN  yarn install --frozen-lockfile && cd ./packages/jaeger-ui && yarn build
   8 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c yarn install --frozen-lockfile && cd ./packages/jaeger-ui && yarn build" did not complete successfully: exit code: 1

```

### CODE REVIEW

**Review by lukastaegert — COMMENTED** [review]
There are still some CI issues, please have a look

**Comment by pavolloffay** [comment]
@lukastaegert could you please run the CI? I have pushed one commit to fix the issue.

**Comment by pavolloffay** [comment]
@lukastaegert could you please run the CI again? :)

**Comment by pavolloffay** [comment]
@lukastaegert I have filed https://github.com/rollup/rollup/issues/5354 for wasm-node to unblock our build. Any help is much appreciated

**Comment by pavolloffay** [comment]
@lukastaegert could you please run the CI again?

**Comment by lukastaegert** [comment]
To my limited knowledge, these kind of errors usually mean that it tries to use the wrong linker (it actually says "linking with cc failed"). One solution sometimes is to use "zig", otherwise we need to try stuff. Cross-compilation is apparently non-trivial.

**Comment by pavolloffay** [comment]
Do you have any examples of using the "zig"?

**Comment by pavolloffay** [comment]
@lukastaegert could you please enable the CI again :)

**Comment by pavolloffay** [comment]
`zig` does not seem to work 

```
Internal Error: powerpc64le-unknown-linux-gnu can not be cross compiled by zig
    at BuildCommand.<anonymous> (/home/runner/work/rollup/rollup/node_modules/@napi-rs/cli/scripts/index.js:11446:27)
    at Generator.next (<anonymous>)
    at /home/runner/work/rollup/rollup/node_modules/@napi-rs/cli/scripts/index.js:3552:69
    at new Promise (<anonymous>)
    at __awaiter$1 (/home/runner/work/rollup/rollup/node_modules/@napi-rs/cli/scripts/index.js:3548:10)
    at BuildCommand.execute (/home/runner/work/rollup/rollup/node_modules/@napi-rs/cli/scripts/index.js:11325:16)
    at BuildCommand.validateAndExecute (/home/runner/work/rollup/rollup/node_modules/@napi-rs/cli/scripts/index.js:[21](https://github.com/rollup/rollup/actions/runs/7617255446/job/20757575020?pr=5350#step:12:22)13:37)
    at /home/runner/work/rollup/rollup/node_modules/@napi-rs/cli/scripts/index.js:3154:53
    at noopCaptureActivator (/home/runner/work/rollup/rollup/node_modules/@napi-rs/cli/scripts/index.js:3380:12)
    at Cli.run (/home/runner/work/rollup/rollup/node_modules/@napi-rs/cli/scripts/index.js:3154:30)
```

Any help would be appreciated. Now I am stuck again.

**Comment by lukastaegert** [comment]
It seems we need a profoundly new cross-compilation process. I am no expert either (otherwise, I might just have fixed things). Here https://github.com/cross-rs/cross they claim to be able to compile for those targets, one we need to figure out if this can be married to napi-rs. Or maybe there are other ways to do cross-compilation.

**Comment by uweigand** [comment]
Given the success we've had with s390x here https://github.com/rollup/rollup/pull/5346, I think the same approach ought to work for ppc64le as well.  This means:
- removing the `zig: true` line again; and
- setting the cargo linker environment variable before the npm build call, something like
```
              export CARGO_TARGET_POWERPC64LE_UNKNOWN_LINUX_GNU_LINKER=powerpc64le-linux-gnu-gcc
```

**Review by lukastaegert — APPROVED** [review]
So I used my maintainer privileges and pushed the proposed changes and alas, IT WORKED! This looks good from my side now, and unless there are any concerns, I will merge and release this later today.

**Comment by Swapnali911** [comment]
> Given the success we've had with s390x here #5346, I think the same approach ought to work for ppc64le as well. This means:
> 
> * removing the `zig: true` line again; and
> * setting the cargo linker environment variable before the npm build call, something like
> 
> ```
>               export CARGO_TARGET_POWERPC64LE_UNKNOWN_LINUX_GNU_LINKER=powerpc64le-linux-gnu-gcc
> ```

Thanks @uweigand  I have tried cross build for ppc64le and it is working as expected

**Comment by uweigand** [comment]
Thanks again, @lukastaegert !   Will this be in the 4.13.2 release then?

**Comment by lukastaegert** [comment]
There were some issues with the folder name but it is released now and hopefully works. Apparently, Node calls it ppc64le while Rust likes the longer powerpc64le... I hope I resolved it correctly.

**Comment by uweigand** [comment]
Ah, I see.  This looks good to me ...

File: native.js
Comment (sapphi-red) [inline native.js:19]: @pavolloffay @lukastaegert 
I guess this `ppc64le` and `cpu` field in package.json need to be `ppc64`. `process.arch` seems to return `ppc64` instead of `ppc64le`.
https://github.com/konveyor/tackle2-ui/actions/runs/8150208242/job/22276088934#step:7:89

Comment (uweigand) [reply native.js:19]: Looks like you are right and `process.arch` always returns `ppc64` - which in practice means ppc64le as Node hasn't been supporting big-endian Linux for a while.  In theory there is also an `os.machine` interface that returns either `ppc64` or `ppc64le`, but it doesn't seem necessary to use that.

Comment (lukastaegert) [reply native.js:19]: If anyone wants to make an educated PR to improve here, this would be very welcome! Otherwise, I can also have a look in the next days.

Comment (lukastaegert) [reply native.js:19]: PR here: #5460

Comment (uweigand) [reply native.js:19]: Hi @sapphi-red, just wondering whether the native ppc64le rollup work fors you now?  It looks like tackle2-ui downgraded to rollup v3 in the meantime?

Comment (sapphi-red) [reply native.js:19]: I don't have a ppc64le machine so I don't know 😅

Comment (uweigand) [reply native.js:19]: I see, thanks!

Comment (lukastaegert) [reply native.js:19]: I spend some hours setting up emulated smoke tests for some environments including ppc64le and I can confirm it works now 😀
https://github.com/rollup/rollup/pull/5477

**Pull request merged**
